File: server/persistence/errors_test.go
// Copyright 2020 - Offen Authors <hioffen@posteo.de>
// SPDX-License-Identifier: Apache-2.0

package persistence

import "testing"

func TestErrUnknownAccount(t *testing.T) {
	err := ErrUnknownAccount("unknown")
	if message := err.Error(); message != "unknown" {
		t.Errorf("Unexpected error message %s", message)
	}
}

func TestErrUnknownSecret(t *testing.T) {
	err := ErrUnknownSecret("unknown")
	if message := err.Error(); message != "unknown" {
		t.Errorf("Unexpected error message %s", message)
	}
}

File: auditorium/src/action-creators/errors.test.js
/**
 * Copyright 2020 - Offen Authors <hioffen@posteo.de>
 * SPDX-License-Identifier: Apache-2.0
 */

const assert = require('assert')

const errors = require('./errors')

describe('src/action-creators/errors.js', function () {
  describe('formValidation(error)', function () {
    it('creates a form validation error action', function () {
      const action = errors.formValidation(new Error('Not valid!'))
      assert.deepStrictEqual(action, {
        type: 'FORM_VALIDATION_ERROR',
        payload: {
          flash: 'Not valid!'
        }
      })
    })
  })

  describe('unrecoverable(error)', function () {
    it('creates a unrecoverable error action', function () {
      const action = errors.unrecoverable(new Error('Did not work!'))
      assert.strictEqual(action.type, 'UNRECOVERABLE_ERROR')
      assert.strictEqual(action.payload.message, 'Did not work!')
    })
  })
})
